steward/src/cellars.rs

```diff
@@ -228,14 +236,16 @@ pub fn validate_cache_price_router(
     cellar_id: &str,
     check_total_assets_value: bool,
     allowable_range_value: u32,
+    expected_price_router: Option<&str>,
 ) -> Result<(), Error> {
-    if !check_total_assets_value || allowable_range_value != 500 {
-        return Err(sp_call_error(
-            "unauthorized arguments for cachePriceRouter call".to_string(),
-        ));
-    }
     let cellar_id_normalized = normalize_address(cellar_id.to_string());
-    if !ALLOWED_CACHE_PRICE_ROUTER.contains(&cellar_id_normalized.as_str()) {
+
+    if !ALLOWED_CACHE_PRICE_ROUTER.iter().any(|(cellar_id, args)| {
+        cellar_id_normalized.eq(cellar_id)
+            && args.0 == check_total_assets_value
+            && args.1 >= allowable_range_value
```

Comment:
I believe this is meant to be `<=` rather than `>=`.

Reply:
`args.1` is the upper limit, `allowable_range_value` is the argument. maybe a confusing variable name. the upper limit has to be the same or higher than the passed in value.

Reply:
Might make it clearer if we refer to the hardcoded values as permissions rather than args.